go.mod

```diff
@@ -10,14 +10,14 @@ require (
 	github.com/open-policy-agent/cert-controller v0.3.0
 	github.com/prometheus/client_golang v1.12.1
 	go.uber.org/zap v1.21.0
-	k8s.io/api v0.23.4
-	k8s.io/apimachinery v0.23.4
-	k8s.io/client-go v0.23.4
-	k8s.io/component-base v0.23.3
-	k8s.io/component-helpers v0.23.4
-	k8s.io/klog/v2 v2.40.1
+	k8s.io/api v0.24.0
+	k8s.io/apimachinery v0.24.0
+	k8s.io/client-go v0.24.0
+	k8s.io/component-base v0.24.0
+	k8s.io/component-helpers v0.24.0
+	k8s.io/klog/v2 v2.60.1
```

Comment:
this is not 0.24, so this PR is updating more than a single dependency, the title of the PR says 0.24 (and doesn't mention of what) 

Please enhance your PR description and Git commit to reflect all these changes